cluster/admin-sessions.go

```diff
@@ -36,19 +36,30 @@ type AdminSession struct {
 	WhoAmI           string
 }
 
-func CreateAdminSession(ctx *Context, adminID *uuid.UUID) (*AdminSession, error) {
+func CreateAdminSession(ctx *Context, adminID *uuid.UUID, idpSession bool, expirationDate time.Time) (*AdminSession, error) {
```

Comment:
Can this be probably two separate functions for each behavior?
Since if idpSession is false it won't consider the mandatory input parameter expirationDate.